runtime/src/info.rs

```diff
@@ -25,3 +28,55 @@ impl FwInfoCmd {
         }))
     }
 }
+
+pub struct IDevIdInfoCmd;
+impl IDevIdInfoCmd {
+    pub(crate) fn execute(drivers: &Drivers) -> CaliptraResult<MailboxResp> {
+        let pub_key = drivers.fht.idev_dice_pub_key;
+
+        Ok(MailboxResp::GetIdevInfo(GetIdevInfoResp {
+            hdr: MailboxRespHeader::default(),
+            idev_pub_x: pub_key.x.into(),
+            idev_pub_y: pub_key.y.into(),
+        }))
+    }
+}
+
+pub struct IDevIdCsrCmd;
+impl IDevIdCsrCmd {
+    pub(crate) fn execute(drivers: &Drivers, cmd_args: &[u8]) -> CaliptraResult<MailboxResp> {
+        if let Some(cmd) = GetIdevCsrReq::read_from(cmd_args) {
+            const SIGNATURE_SIZE: usize = 96;
+
+            let Ok(in_len) = usize::try_from(cmd.data_size) else {
+                return Err(CaliptraError::RUNTIME_MAILBOX_INVALID_PARAMS);
+            };
+
+            if in_len > cmd.data.len() {
+                return Err(CaliptraError::RUNTIME_MAILBOX_INVALID_PARAMS);
+            }
+            let mut resp_data = [0; GetIdevCsrResp::DATA_MAX_SIZE];
+            resp_data[..in_len].copy_from_slice(&cmd.data[..in_len]);
```

Comment:
X.509 encoding wraps the TBS and signature in an additional header and includes algorithm information along with the signature, so it's not sufficient to just concatenate them.

Instead, you can use [Ecdsa384CertBulder](https://github.com/chipsalliance/caliptra-sw/blob/8b23c0a6d1fb727c932d2f84b9ca4aa543214da8/x509/src/cert_bldr.rs#L187) and call `.build` which will do the proper X.509 encoding.